pkg/probes/legacy/legacy_test.go

```diff
@@ -213,7 +213,7 @@ func TestLegacyProbe_GetExpandedUserData(t *testing.T) {
 
 			prb := Probe{}
 			// First check if function is returning an error
-			got, err := prb.GetExpandedUserData(tt.userDataVariables)
+			got, err := prb.GetExpandedUserData(tt.userDataVariables, userDataTemplate)
```

Comment:
Consider passing an empty string here instead of userdataTemplate to reduce potential confusion